memo_app.rb

```diff
@@ -3,37 +3,60 @@
 
 require 'sinatra'
 require 'sinatra/reloader'
-require 'json'
 require 'byebug'
 require 'cgi'
+require 'pg'
 
-def collect_memo_data
-  file_names = Dir.glob('*', base: './memos')
-  @memo_files = file_names.map do |name|
-    File.open("./memos/#{name}") do |file|
-      read_line = file.read
-      JSON.parse(read_line)
-    end
-  end
+def make_a_connection_to_db
+  host = 'localhost'
+  port = 5432
+  db = 'memodb'
+  user = 'memoapp'
+  PG::Connection.new(host: host, port: port, dbname: db, user: user)
 end
 
-def sort_by_created_at
-  @memo_files = @memo_files.sort_by { |a| a['created_at'] }
-end
+Connection = make_a_connection_to_db
 
-def reverse_memos_order
-  @memo_files = @memo_files.reverse
-end
+class Memo
+  attr_reader :id, :title, :body, :created_at, :edited_at
+
+  def initialize(hash)
+    @id = hash['id']
+    @title = hash['title']
+    @body = hash['body']
+    @created_at = hash['created_at']
+    @edited_at = hash['edited_at']
+  end
+
+  def self.create_a_memo_array
+    all_memo_data = Connection.exec('SELECT * FROM memos').to_a
+    all_memo_data.map { |hash| Memo.new(hash) }.sort_by(&:created_at).reverse
+  end
+
+  def self.find_a_memo(id)
+    memo_data = Connection.exec('SELECT * FROM memos WHERE id = $1', [id]).to_a
+    memo = memo_data.map { |hash| Memo.new(hash) }
+    memo[0]
+  end
+
+  def self.post_a_memo(title_params, body_params, created_at_params)
+    Connection.prepare('create', 'INSERT INTO memos (title, body, created_at) VALUES ($1, $2, $3)')
+    Connection.exec_prepared('create', [title_params, body_params, created_at_params])
+  end
 
-def find_selected_memo(number)
-  collect_memo_data
-  @selected_memo = @memo_files.find { |file| file['number'] == number.to_i }
+  def self.edit_a_memo(new_title, new_body, edited_at, id)
+    Connection.prepare('update', 'UPDATE memos SET title = $1, body = $2, edited_at = $3 WHERE id = $4')
+    Connection.exec_prepared('update', [new_title, new_body, edited_at, id])
+  end
+
+  def self.delete_a_memo(id)
+    Connection.prepare('delete', 'DELETE from memos WHERE id = $1')
```

Comment:
ここだけ SQL の FROM が小文字になっているようです 👀 

```suggestion
    Connection.prepare('delete', 'DELETE FROM memos WHERE id = $1')
```